tests/test_tracking_utils.py

```diff
@@ -58,10 +59,42 @@
             "train-finetune-student-ru-en",
             ("finetune-student", None, None, None),
         ),
+        (
+            "train-teacher-ru-en-1",
+            ("teacher-1", None, None, None),
+        ),
+        (
+            "evaluate-backward-url-gcp_pytest-dataset_a0017e-en-ru",
+            ("backward", "url", "gcp_pytest-dataset_a0017e", None),
+        ),
+        (
+            "train-teacher-ast-en-1",
+            ("teacher-1", None, None, None),
+        ),
+        (
+            # Test the 3-letter language codes like "Asturian".
+            "evaluate-student-sacrebleu-wmt19-ast-en",
+            ("student", "sacrebleu", "wmt19", None),
+        ),
     ],
 )
-def test_parse_tag(example, parsed_values):
-    assert parse_tag(example) == parsed_values
+def test_parse_task_label(task_label, parsed_values):
+    assert parse_task_label(task_label) == ParsedTaskLabel(*parsed_values)
+
+
+def test_parse_labels_on_full_taskgraph():
+    """Ensure that all the taskgraph task labels parse."""
+    for task in get_full_taskgraph():
+        if not task.startswith("train-") and not task.startswith("evaluate-"):
```

Comment:
nice, we'll catch all the issues in the tests now. Why are you skipping train- and evaluate-? Can we just enable this test for everything?

Reply:
The logic is inverted here, so if it's not train and not evaluate, it's skipping. Maybe I should rewrite it as:

`if not (task.startswith("train-") or task.startswith("evaluate-")):`